src/sentry_tracing.c

```diff
@@ -151,6 +154,26 @@ sentry_transaction_context_remove_sampled(sentry_transaction_context_t *tx_cxt)
     }
 }
 
+///
+/// @param s given string
+/// @param len the length of the string
+/// @return whether the string is a valid hex string of at least the given
+///         length which contains at least one non-zero character
+bool
+is_valid_nonzero_hexstring(const char *s, size_t len)
+{
+    bool has_nonzero = false;
+    for (size_t i = 0; i < len; i++) {
+        if (!isxdigit(s[i])) {
+            return false;
+        }
+        if (s[i] != '0') {
+            has_nonzero = true;
```

Comment:
If the condition is to have _at least one_ non-zero character, you could early return here as well.

Reply:
I think we want at least one non-zero character **and** for all characters to be hexadecimal, so seeing one non-zero character isn't enough (e.g. we could have `00000000ABCZZZZZ` where Z makes it an invalid span-id).